apis/apps/v1alpha1/backupschedule_types.go

```diff
@@ -20,54 +20,88 @@ import (
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
 
-type BackupSchedule struct {
+type NebulaScheduledBackup struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   BackupScheduleSpec   `json:"spec,omitempty"`
-	Status BackupScheduleStatus `json:"status,omitempty"`
+	Spec   ScheduledBackupSpec   `json:"spec,omitempty"`
+	Status ScheduledBackupStatus `json:"status,omitempty"`
 }
 
 // +kubebuilder:object:root=true
-// BackupScheduleList contains a list of BackupSchedule.
-type BackupScheduleList struct {
+// NebulaScheduledBackupList contains a list of NebulaScheduledBackup.
+type NebulaScheduledBackupList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 
-	Items []BackupSchedule `json:"items"`
+	Items []NebulaScheduledBackup `json:"items"`
 }
 
-// BackupScheduleSpec contains the specification for a backupSchedule of a nebula cluster backupSchedule.
-type BackupScheduleSpec struct {
+// ScheduledBackupSpec contains the specification for a NebulaScheduledBackup of a nebula cluster NebulaScheduledBackup.
+type ScheduledBackupSpec struct {
 	// Schedule specifies the cron string used for backup scheduling.
 	Schedule string `json:"schedule"`
 	// Pause means paused backupSchedule
-	Pause bool `json:"pause,omitempty"`
-	// MaxBackups is to specify how many backups we want to keep
-	// 0 is magic number to indicate un-limited backups.
-	// if MaxBackups and MaxReservedTime are set at the same time, MaxReservedTime is preferred
+	Pause *bool `json:"pause,omitempty"`
+	// MaxBackups specifies how many backups we want to keep in the remote storage bucket.
+	// 0 is the magic number to indicate unlimited backups.
+	// if both MaxBackups and MaxReservedTime are set at the same time, MaxReservedTime will be used
 	// and MaxBackups is ignored.
 	MaxBackups *int32 `json:"maxBackups,omitempty"`
-	// MaxReservedTime is to specify how long backups we want to keep.
-	MaxReservedTime *string `json:"maxReservedTime,omitempty"`
-	// BackupTemplate is the specification of the backup structure to get scheduled.
+	// MaxRetentionTime specifies how long we want the backups in the remote storage bucket to be kept for.
+	// +kubebuilder:validation:Pattern=`^([0-9]+(\.[0-9]+)?(s|m|h))+$`
+	MaxRetentionTime *string `json:"maxRetentionTime,omitempty"`
+	// BackupTemplate is the specification of the backup structure to schedule.
 	BackupTemplate BackupSpec `json:"backupTemplate"`
-	// LogBackupTemplate is the specification of the log backup structure to get scheduled.
+	// MaxSuccessfulNublaBackupJobs specifies the maximum number of successful backup jobs to keep. Default 3.
+	MaxSuccessfulNebulaBackupJobs *int32 `json:"maxSuccessfulNebulaBackupJobs,omitempty"`
+	// MaxFailedNebulaBackupJobs specifies the maximum number of failed backup jobs to keep. Default 3
+	MaxFailedNebulaBackupJobs *int32 `json:"maxFailedNebulaBackupJobs,omitempty"`
```

Comment:
Why are these fields necessary? I believe users aren't particularly concerned with how many NebulaBackups need to be retained. This is precisely the task for NebulaScheduled. It determines retention based on MaxBackups and MaxRetentionTime.

Additionally, I think the lifecycle of NebulaBackup should be responsible for determining when the corresponding storage is deleted. In other words, when NebulaBackup is deleted, the corresponding storage should also be deleted, maintaining a consistent lifecycle.

NebulaScheduled only operates on NebulaBackup, deleting relevant resources when NebulaBackup is removed.

Reply:
Yes, MaxBackups and MaxRetentionTime is for determining how many backups to keep in either the S3 bucket or the gcp bucket. These parameters are there to specify how many backup jobs the user wants to keep as the user may want to keep the backup itself but not the actual job that created it. This also allows the user to keep all failed backup logs if needed. @MegaByte875 what do you think?